app/resources/snippet_resource.rb

```diff
@@ -0,0 +1,45 @@
+class SnippetResource < JSONAPI::Resource
+  attributes :content
+  has_one :user
+
+  def initialize(current_user, snippets)
+    @current_user = current_user
+    @snippets = Array(snippets)
+  end
+
+  def custom_json
```

Comment:
We shouldn't be building ad-hoc JSON objects in code.  This should be handled by a serializer.  That way, when you need to change the schema, you touch only the model and the serializer class to change payloads.